File: 백준/Silver/9237. 이장님 초대/이장님 초대.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    int numOfSeedlings;

    cin >> numOfSeedlings;

    vector<int> daysForGrowth;

    for (int i = 0; i < numOfSeedlings; i++) {
        int input;
        cin >> input;

        daysForGrowth.push_back(input);
    }

    // 내림차순 정렬
    sort(daysForGrowth.begin(), daysForGrowth.end(), greater<int>());

    for (int i = 0; i < numOfSeedlings; i++) {
        daysForGrowth[i] = daysForGrowth[i] + i + 1;
    }

    int max = 0;

    for (int i = 0; i < numOfSeedlings; i++) {
        if (max < daysForGrowth[i]) {
            max = daysForGrowth[i];
        }
    }

    cout << max + 1;

    return 0;
}

File: 백준/Silver/9237. 이장님 초대/이장님 초대.java
import java.util.Arrays;
import java.util.Collections;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int numOfSeedlings;

        numOfSeedlings = sc.nextInt();

        Integer[] daysForGrowth = new Integer[numOfSeedlings];

        for (int i = 0; i < numOfSeedlings; i++) {
            int input;
            input = sc.nextInt();

            daysForGrowth[i] = input;
        }

        // 내림차순 정렬
        Arrays.sort(daysForGrowth, Collections.reverseOrder());

        for (int i = 0; i < numOfSeedlings; i++) {
            daysForGrowth[i] = daysForGrowth[i] + i + 1;
        }

        int max = 0;

        for (int i = 0; i < numOfSeedlings; i++) {
            if (max < daysForGrowth[i]) {
                max = daysForGrowth[i];
            }
        }

        System.out.print(max + 1);

        sc.close();
    }
}
